### DISCUSSION TITLE
Device Initialization Failed

### DISCUSSION THREAD

**Discussion by jc2019211088** [PicoQuant/snAPI#39] [Q&A] [answered]
Hi, I am trying to use the snAPI for control the picoharp 300.While using snAPI, I followed the instructions in the start menu and replaced the ini file with 330cfd.ini, but I still encountered a bug where the device initialization failed. Could you please tell me if there's a step that I might have missed? Alternatively, could you inform me of the additional steps required for initializing the device?
![图片11](https://github.com/user-attachments/assets/fab6a077-8d23-4003-83f5-8c7ed84f6bca)

**Comment by snizzleorg** [comment]
Could you please let us know whether your device is a PicoHarp300 or a PicoHarp330?

Snapi is only compatible with the latter.

**Reply by jc2019211088** [reply]
Yes, the device I'm using is PH330, which has four channels. However, after running it on the terminal, there is a mismatch and extra channels appear. Could it be that my API version is incorrect?

![图片3](https://github.com/user-attachments/assets/d224ce5f-2ee2-44a8-ae72-455235fdd2df)

**Reply by snizzleorg** [reply]
Does the PicoHarp330 work without any problems when you use the PicoHarp330 software?

**Reply by tpoint75** [reply, marked as answer]
Oh, I am sorry, the 4 Channel PH330 was updated to the driver version v2.0
Please install this software (and the containing driver) https://www.picoquant.com/dl_software/PicoHarp330/PicoHarp330_SW_and_DLL_v2_0.zip
